checker/src/org/checkerframework/checker/index/upperbound/UpperBoundVisitor.java

```diff
@@ -128,32 +128,80 @@ protected void commonAssignmentCheck(
         // Either a singleton list of the single integer on the right,
         // or a list containing all the values in a constant array.
         List<Integer> rhsValues = new ArrayList<>();
-        // All the values in the must be compile-time constants.
```

Comment:
I've cleaned this up as best I can. It's still pretty bad, but I don't see any reasonable alternatives beyond what I've already done.